front/app/utils/permissions/permissions.ts

```diff
@@ -16,25 +16,20 @@ interface IResourceData {
   [key: string]: any;
 }
 
-interface IPermissionRule {
-  (
-    resource: TPermissionItem | null,
-    user: IUser | null,
-    tenant: IAppConfigurationData,
-    context?: any
-  ): boolean;
-}
-
-interface IPermissionRules {
-  [key: string]: {
-    [key: string]: IPermissionRule;
-  };
-}
+type PermissionRule = (
+  resource: TPermissionItem | null,
+  user: IUser | undefined,
```

Comment:
Only "big" change: user type change. Can be undefined because we're now passing user instead of falling back to null when user is undefined.